ios/testmanagerd/testlistener_test.go

```diff
@@ -0,0 +1,170 @@
+
+import (
+	"io"
+	"os"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/danielpaulus/go-ios/ios/nskeyedarchiver"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFinishExecutingTestPlan(t *testing.T) {
+	t.Parallel()
+
+	t.Run("Wait for test finish with single waiter", func(t *testing.T) {
+		testListener := NewTestListener(os.Stdout, os.Stdout)
+
+		go func() {
+			testListener.didFinishExecutingTestPlan()
+		}()
+
+		<-testListener.testFinishedChannel
+	})
+
+	t.Run("Wait for test finish with multiple waiters", func(t *testing.T) {
+		testListener := NewTestListener(os.Stdout, os.Stdout)
+
+		var wg sync.WaitGroup
+		wg.Add(2)
+
+		testListener.didFinishExecutingTestPlan()
+
+		go func() {
+			<-testListener.Done()
+			wg.Done()
+		}()
+
+		go func() {
+			<-testListener.Done()
+			wg.Done()
+		}()
+
+		wg.Wait()
+	})
+
+	t.Run("Check error on a failed test run", func(t *testing.T) {
+		testListener := NewTestListener(os.Stdout, os.Stdout)
+
+		testListener.initializationForUITestingDidFailWithError(nskeyedarchiver.NSError{
+			ErrorCode: 1, Domain: "testdomain", UserInfo: map[string]interface{}{}})
+
+		<-testListener.Done()
+		assert.Error(t, testListener.err)
+	})
+
+	t.Run("Check error on a failed test run with bootstrap error", func(t *testing.T) {
+		testListener := NewTestListener(os.Stdout, os.Stdout)
+
+		testListener.didFailToBootstrapWithError(nskeyedarchiver.NSError{
+			ErrorCode: 1, Domain: "testdomain", UserInfo: map[string]interface{}{}})
+
+		<-testListener.Done()
+		assert.Error(t, testListener.err)
+	})
+
+	t.Run("Check test log callbacks", func(t *testing.T) {
+		logWriter := assertionWriter{}
+		debugLogWriter := assertionWriter{}
+
+		testListener := NewTestListener(&logWriter, &debugLogWriter)
+
+		testListener.LogMessage("log")
+		testListener.LogDebugMessage("debug")
+
+		assert.True(t, logWriter.hasBytes, "Test listener must receive test logs")
+		assert.True(t, debugLogWriter.hasBytes, "Test listener must receive test debug logs")
+	})
+
+	t.Run("Check test suite creation", func(t *testing.T) {
+		testListener := NewTestListener(io.Discard, io.Discard)
+
+		testListener.testSuiteDidStart("mysuite", "2024-01-16 15:36:43 +0000")
+
+		assert.Equal(t, 2024, testListener.testSuite.StartDate.Year())
+		assert.Equal(t, time.Month(1), testListener.testSuite.StartDate.Month())
+		assert.Equal(t, 16, testListener.testSuite.StartDate.Day())
+		assert.Equal(t, "mysuite", testListener.testSuite.Name)
+	})
+
+	t.Run("Check test case creation", func(t *testing.T) {
+		testListener := NewTestListener(io.Discard, io.Discard)
+
+		testListener.testSuiteDidStart("mysuite", "2024-01-16 15:36:43 +0000")
+		testListener.testCaseDidStartForClass("myclass", "mymethod")
+
+		assert.Equal(t, 1, len(testListener.testSuite.TestCases), "TestCase must be appended to list of test cases")
+		assert.Equal(t, "myclass", testListener.testSuite.TestCases[0].ClassName)
+		assert.Equal(t, "mymethod", testListener.testSuite.TestCases[0].MethodName)
```

Comment:
we could check everything related to a single testcase at once
```suggestion
		assert.Equal(t, TestCase{
			ClassName:  "myclass",
			MethodName: "mymethod",
		}, testListener.testSuite.TestCases[0])
```

Reply:
Didn't know that, thank you.

Reply:
Applied